src/components/filter_configurator.js

```diff
@@ -0,0 +1,102 @@
+const React = require('react')
+const { connect } = require('react-redux')
+const { PropTypes } = React
+const Dialog = require('material-ui/Dialog').default
+const FlatButton = require('material-ui/FlatButton').default
+const { List, ListItem } = require('material-ui/List')
+const Toggle = require('material-ui/Toggle').default
+const {defineMessages, injectIntl, intlShape} = require('react-intl')
+
+const getCandidateFilters = require('../selectors/candidate_filters')
+const { addVisibleFilter, removeVisibleFilter } = require('../action_creators')
+
+
+const messages = defineMessages({
+  configureFilters: {
+    id: 'filter.configure',
+    defaultMessage: 'Configure Filters',
+    description: 'Dialog title text when displaying available filters'
+  }
+})
+
+class FilterConfigurator extends React.Component {
+  static propTypes = {
+    candidateFilters: PropTypes.array.isRequired,
+    handleClose: PropTypes.func.isRequired,
+    intl: intlShape.isRequired,
+    onAddVisibleFilter: PropTypes.func.isRequired,
+    onRemoveVisibleFilter: PropTypes.func.isRequired,
+    visibleFilters: PropTypes.array.isRequired
+  }
+
+  onToggle = (fieldName) => {
+    const { onAddVisibleFilter, onRemoveVisibleFilter } = this.props
+    return ({target}) => {
+      if (target.checked) {
+        onAddVisibleFilter(fieldName)
+      } else {
+        onRemoveVisibleFilter(fieldName)
+      }
+    }
+  }
+
+  render () {
+    const { candidateFilters, handleClose, intl, visibleFilters } = this.props
+
+    const actions = [
+      <FlatButton
+        label='Done'
+        primary
+        onTouchTap={handleClose}
+      />
+    ]
+
+    // FormattedMessage
+    return (
+      <Dialog
+        title={intl.formatMessage(messages.configureFilters)}
```

Comment:
Just `<FormattedMessage>` renders a <span> because of the material-ui "bug" but we can just do `<h3><FormattedMessage {...messages.configureFilters} /></h3>` and style the h3 as we like, which I think is neater, and avoids the need to inject intl